rust/src/store/block_store_impl.rs

```diff
@@ -991,22 +991,22 @@ impl BlockStore for IndexerStore {
 /// `{block height BE}{state hash}`
 fn block_height_key(block: &PrecomputedBlock) -> Vec<u8> {
     let mut key = to_be_bytes(block.blockchain_length());
-    key.append(&mut block.state_hash().to_bytes());
+    key.append(&mut block.state_hash().to_bytes().to_vec());
```

Comment:
If we almost always have to convert it to a `Vec`, there might not be much savings by avoiding a `Vec` to begin with.